### PR TITLE
fix for "Mounted" event

### PR DESCRIPTION
**Pull request by seidhrhealz**
now manually observes mounted status to fix the unreliable Mounted event

### CODE REVIEW

File: ItemRack/ItemRackEvents.lua [hunk 1]
```diff
@@ -368,6 +368,18 @@ function ItemRack.ProcessZoneEvent()
 	end
 end
 
+--here we observe mounted status and raise an event should it change. UNIT_AURA event seems unreliable for this
+local _playerMounted = IsMounted() and not UnitOnTaxi("player")
```
Comment (Rottenbeer) [inline ItemRack/ItemRackEvents.lua]: Please find a better name for _playerMounted, maybe _playerLastMountState?

Comment (seidhrhealz) [reply ItemRack/ItemRackEvents.lua]: I have now renamed it, however not to _playerLastMountState which I believe would make sense only if actually saved the state (_playerLastMountState = "MOUNTED", _playerLastMountState = "DISMOUNTED") and not use the variable as a boolean. With there being only 2 states I am happy with a boolean.

**Comment by seidhrhealz** [comment]
> Did you test that it works reliable now? I don't have access to a WoW account anymore.
> I'm unsure if adding a 0.5 seconds timer that checks if you are mounted or not is the way to go, but if should work more reliable than UNIT_AURA

I have been using this for a while already in ItemRackAutoRidingGear as well as ItemRack itself for the past week or so. It seems to be working fine and I have tested all possible scenarios I could come up with that relate to mounting/dismounting.

**Comment by seidhrhealz** [comment]
> Please change the commit message to:
> "Add a timer to check for "Mounted" events"

Last commit message is now as specified above.

**Comment by seidhrhealz** [comment]
squashed

**Comment by seidhrhealz** [comment]
> 
> 
> Looks fine now, please squash the commits and I'll take it

squashed

**Comment by Rottenbeer** [comment]
Thanks!

**Pull request merged**